xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -9126,4 +9169,71 @@ public int getLocalReferenceMaxLength()
     {
         return getStore().getLimitSize(this.getXWikiContext(), this.getClass(), "fullName");
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public DocumentAuthors getAuthors()
+    {
+        if (this.authors == null) {
+            this.authors = new DefaultDocumentAuthors();
+        }
+        return this.authors;
+    }
+
+    /**
+     * Allow to set the different authors of the document.
+     * Note that if one of the author changed as part of this call, the {@link #setMetaDataDirty(boolean)} flag is set
+     * to {@code true}.
+     *
+     * @param authors the authors information.
+     * @since 14.0RC1
+     */
+    @Unstable
+    public void setAuthors(DocumentAuthors authors)
```

Comment:
I don't think setAuthors is really needed now that DocumentAuthors has setters.

Reply:
The immediate UC I can think of for this setter, is if someone needs for some reason another implem of DocumentAuthors, then it's handy to be able to set the instance.

Reply:
I kept it, but it only copies the value of the given parameter.

Reply:
What about a `DocumentAuthors#set(DocumentAuthors)` instead ?